src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/MiscStatement.scala

```diff
@@ -159,4 +160,33 @@ private[querygeneration] object MiscStatement {
       case DoubleType => "DOUBLE"
       case _ => null
     })
+
+  /** For known unsupported data conversion, raise exception to break the pushdown process.
+    * For example, snowflake doesn't support to convert DATE/TIMESTAMP to NUMBER
+    */
+  private[querygeneration] def raiseExceptionIfCastIsNotSupported(srcType: DataType, targetType: DataType): Unit = {
+    val isSupported = srcType match {
+      case _: DateType | _: TimestampType =>
+        targetType match {
+          case _: DecimalType | _: IntegerType | _: LongType | _: FloatType | _: DoubleType =>
+            // snowflake doesn't support to convert DATE/TIMESTAMP to NUMBER
+            false
+          case _ => true
+        }
+      case _ => true
+    }
+
+    if (!isSupported) {
+      // This exception will not break the connector. It will be caught in
+      // QueryBuilder.treeRoot and a telemetry message will be sent if
+      // there are any snowflake tables in the query.
+      throw new SnowflakePushdownUnsupportedException(
+        SnowflakeFailMessage.FAIL_PUSHDOWN_UNSUPPORTED_CONVERSION,
+        s"Don't support to convert $srcType column to $targetType type",
+        "",
+        false
+      )
+    }
+  }
```

Comment:
Can be simplified to two lines
```scala
(srcType, targetType) match {
case (DateType, NumericType) => throw Exception
case (TimestampType, NumericType) => throw Exception
}
```

Reply:
And I don't think you need this separate function, move it inline

Reply:
I just tried, but it looks `NumericType` is spark private. SC can't access it.
`private[spark] object NumericType extends AbstractDataType {...}`

Reply:
If it can't be simplified as you mentioned, I think it is better to use a separate function.

Reply:
it is simplified as below:
```
      case Cast(child, t, _) =>
        getCastType(t) match {
          case Some(cast) =>
            /** For known unsupported data conversion, raise exception to break the pushdown process.
              * For example, snowflake doesn't support to convert DATE/TIMESTAMP to NUMBER
              */
            (child.dataType, t) match {
              case (_: DateType | _: TimestampType,
              _: IntegerType | _: LongType | _: FloatType | _: DoubleType | _: DecimalType) => {
                // This exception will not break the connector. It will be caught in
                // QueryBuilder.treeRoot and a telemetry message will be sent if
                // there are any snowflake tables in the query.
                throw new SnowflakePushdownUnsupportedException(
                  SnowflakeFailMessage.FAIL_PUSHDOWN_UNSUPPORTED_CONVERSION,
                  s"Don't support to convert ${child.dataType} column to $t type",
                  "",
                  false)
              }
              case _ =>
            }

            ConstantString("CAST") +
              blockStatement(convertStatement(child, fields) + "AS" + cast)
          case _ => convertStatement(child, fields)
        }
```